src/ui/public/url/get_protocol_and_host.js

```diff
@@ -0,0 +1,12 @@
+import { relativeToAbsolute } from './relative_to_absolute';
+import { parse } from 'url';
+
+/**
+ * Returns the protocol and host of the page from where this function is called.
+ * @return {{protocol: {string}, host: {string}}}
+ */
+export function getProtocolAndHost() {
+  const absoluteUrl = relativeToAbsolute('');
```

Comment:
This should probably just be reading `window.location.hostname` `window.location.port` `window.location.protocol`